app/controllers/tokite/rules_controller.rb

```diff
@@ -27,12 +27,17 @@ def edit
   
     def update
       @rule = Rule.find(params[:id])
-      @rule.assign_attributes(rule_params)
-      if @rule.save
-        flash[:info] = "Rule updated."
-        redirect_to user_path(@rule.user_id)
-      else
+      if !SearchQuery.verify_query(rule_params[:query])
```

Comment:
I think tokite should verify query in create action too.
How about implementing this verification as Rule's custom validations? http://guides.rubyonrails.org/active_record_validations.html#custom-methods
tokite already has validation on query.
https://github.com/cookpad/tokite/blob/e41bb64f6313bbb4e4200ae2b5885a35b63ee302/app/models/tokite/rule.rb#L8
https://github.com/cookpad/tokite/blob/e41bb64f6313bbb4e4200ae2b5885a35b63ee302/app/models/tokite/rule.rb#L49-L53